feat(replay): Document how to connect Replay with support software (#9836)
getsentry/sentry-docs: docs/platforms/javascript/common/session-replay/understanding-sessions.mdx

If you return `false` from this method for a given error, we will not check the error sample rate for this error.
If you return `true`, we will continue to check the error sample rate as normal.

### Connect Replays with Support Software

> Replays does not need to be connected to an application error, it can also be used to supplement your support software

Reviewer: ```suggestion
Replays do not need to be connected to an application error, it can also be used to supplement your support tickets
```

> Replays do not need to be connected to an application error, it can also be used to supplement your support tickets
> as well. Depending on the level of customization provided by your support widget, it is possible to send a replay to

Reviewer 2: ```suggestion
Replays do not need to be connected to an application error, they can also be used to supplement your support tickets.
Depending on the level of customization provided by your support widget, it is possible to send a replay to
```

Sentry when your users open the support widget. Note that Sentry's [User Feedback](https://docs.sentry.io/product/user-feedback/#user-feedback-widget)

> provides this functionality by default! You can then send the ID of the replay along with the support ticket that your
> user submits. The ID can then be used to reference the replay inside of Sentry. Alternatively, you can search for
> replays by a user identifier such as email.

Reviewer 2: ```suggestion
provides this functionality by default! You can then send the ID of the replay along with the support ticket that your
user submits. The ID can be used to reference the replay in Sentry. Alternatively, you can search for
replays by a user identifier such as email.
```

> The below is a hypothetical scenario of how you can connect Replay to your support widget.

Reviewer 2: ```suggestion
The example below provides a template for how you might connect Replay to your support widget.
```

```javascript
/**
 * Initialize the Sentry SDK as normal.
 *
 * `replaysOnErrorSampleRate` needs to be > 0 so that replays are always buffering and only sent when necessary
 */
Sentry.init({
  dsn: "...",
  replaysSessionSampleRate: 0.1,
  replaysOnErrorSampleRate: 0.5,
  integrations: [Sentry.replayIntegration()],
});

> /**

Reviewer: Is it intentional that everything is commented out except for the the Replay snippet?

Author: Nope good catch!

 * Your widget will need to have an event/hook to trigger flushing the replay to
 * Sentry.

>  /

Author: ```suggestion
 */
```

MySupportWidget.on('open', () => {
  const replay = Sentry.getReplay();
  // Send replay to Sentry
  replay.flush();
  // Retrieve the replay id that was saved and attach it to your support widget
  const replayId = replay.getReplayId();
  MySupportWidget.setTag('replayId', replayId);
});

/**

>  * Now determine if your support application allows you to use custom data to generate a link directly to the

Reviewer 2: Maybe we can say something like this?

> If your support application allows you to send custom data with the support ticket, you may want to include a link to the replay URL in Sentry. That way, you'll be able to open the replay directly from the ticket itself. Replay URLs have the following format:

>  * Now determine if your support application allows you to use custom data to generate a link directly to the

Reviewer: Do we want to say "..allows you to use custom data to generate a link directly to the replay in Sentry inside your support application"?

Reviewer: I think we're referring to the ability to the show replay url alongside the ticket -- not super clear w this sentence

Author: > Do we want to say "..allows you to use custom data to generate a link directly to the replay in Sentry inside your support application"?

Yeah this is what I'm trying to get at

>  * replay in Sentry. Replay URLs have the following format:

Reviewer: ```suggestion
 * replay in Sentry from within your Support software. Replay URLs have the following format:
```
